src/ConfigCatHOC.tsx

```diff
@@ -24,10 +24,14 @@ export interface WithConfigCatClientProps {
 }
 
 function withConfigCatClient<P>(
-  WrappedComponent: React.ComponentType<P & WithConfigCatClientProps>
+  WrappedComponent: React.ComponentType<P & WithConfigCatClientProps>,
+  configId?: string
 ): React.ComponentType<Omit<P, keyof WithConfigCatClientProps>> {
+
+  const configCatContext = configId ? getOrCreateConfigCatContext(configId) : ConfigCatContext;
+
```

Comment:
The error message "withConfigCatClient must be used..." could be clearer in this case too.

Reply:
Thanks, I updated the error's message generation.